engine/src/main/java/org/terasology/persistence/internal/GamePreviewImageProvider.java

```diff
@@ -0,0 +1,147 @@
+            }
+        }
+        result.removeIf(Objects::isNull);
+        return result;
+    }
+
+    /**
+     * Gets a path to next game preview image.
+     *
+     * @param savePath a path to saves directory
+     * @return a path to preview image
+     */
+    public static Path getNextGamePreviewImagePath(final Path savePath) {
+        final StoragePathProvider storagePathProvider = new StoragePathProvider(savePath);
+        final Path previewsDirPath = storagePathProvider.getPreviewsPath();
+
+        final List<Path> previewsPaths = getAllPathsToFilesInFolder(previewsDirPath);
+        final String fileName = getNextPreviewImageName(previewsPaths);
+
+        return previewsDirPath.resolve(fileName);
+    }
+
+    private static void createDirectory(Path dirPath) {
+        try {
+            Files.createDirectories(dirPath);
+        } catch (IOException e) {
+            logger.warn("Can't create folder, {}", dirPath);
+        }
+    }
+
+    private static List<Path> getAllPathsToFilesInFolder(final Path dirPath) {
+        // create folder if not exists yet
+        createDirectory(dirPath);
+
+        try {
+            return Files.list(dirPath)
+                    .filter(Files::isRegularFile)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            logger.warn("Can't correctly read files from {}", dirPath);
+            return Collections.emptyList();
+        }
+    }
+
+    private static String getNextPreviewImageName(final List<Path> paths) {
+        if (paths.size() < LIMIT) {
+            return getNextNumberFileName(paths);
+        }
+        return getOldestFileName(paths);
+    }
+
+    private static String getOldestFileName(final List<Path> paths) {
+        FileTime oldestTime = null;
+        Path oldestPath = null;
+        for (Path path : paths) {
+            try {
+                FileTime fileTime = Files.getLastModifiedTime(path);
+                if (oldestTime == null || fileTime.compareTo(oldestTime) < 0) {
+                    oldestTime = fileTime;
+                    oldestPath = path;
+                }
+            } catch (IOException e) {
+                logger.error("Can't read last modified time for path {}", path);
+            }
+        }
+
+        if (oldestPath != null) {
+            return oldestPath.getFileName().toString();
+        }
+        return DEFAULT_PREVIEW_NAME;
+    }
+
+    private static String getNextNumberFileName(final List<Path> paths) {
+        int counter = 1;
+        for (Path path : paths) {
+            final String fileName = path.getFileName().toString().split("\\.")[0];
+            try {
+                counter = Math.max(counter, Integer.parseInt(fileName) + 1);
+            } catch (NumberFormatException e) {
+                logger.warn("Could not parse {} as integer (not an error)", fileName, e);
+            }
+        }
+        return counter + JPG_FILE_TYPE;
+    }
+}
```

Comment:
Trivial: newline at the end of every text file

Reply:
Fixed